### DISCUSSION TITLE
[Bug]: Area 93 District 7 (TSML) pulls meeting info from the Antelope Valley Central Office (CF) - TSML is now rejecting the feed from the CF

### DISCUSSION THREAD

**Discussion by CheriJones** [code4recovery/12-step-meeting-list#1523] [Q&A]
### Contact Details

webservant@aascv.org; cherijones@cyberservices.com

### Website With Issue

https://area93district7.org/   https://AVCentraloffice.org

### What happened?

<html><body>
<div>Take your time since all meetings are present. Since the JSON is 
not in the Code for Recovery format, I am not confident they can help. I
 did check the JSON, and I agree it is in a valid JSON format. <div><br></div><div>d</div><div><br></div></div><br><div class="gmail_quote"><div class="gmail_attr">On Fri, Aug 30, 2024 at 6:03 AM &lt;<a href="mailto:cherijones@cyberservices.com" onclick="parent.window.phx.iac.notify('mail_compose', {'to':['cherijones@cyberservices.com']}); return false;" target="_blank">cherijones@cyberservices.com</a>&gt; wrote:<br></div><blockquote class="gmail_quote" style="margin: 0.0px 0.0px 0.0px 0.8ex;border-left: 1.0px solid rgb(204,204,204);padding-left: 1.0ex;">

 
  
  
 
 <div><div style="line-height: 1;padding: 0.5em;">The json feed seems to
 be ok - I am in Montana where they only pretend to have Internet so 
limited as to what I can do - do you know how to post this on GitHub - 
you can go to code <a href="https://deref-mail.com/mail/client/K95h8QXX7HY/dereferrer/?redirectUrl=http%3A%2F%2Frecovery.org" target="_blank">recovery.org</a> for the link<br><br>--<br>Sent with <a href="https://deref-mail.com/mail/client/Wx6gCtsMPRc/dereferrer/?redirectUrl=http%3A%2F%2Fmail.com" target="_blank">mail.com</a> Mail app from my phone</div><div style="line-height: 1;padding: 0.3em;"><div>On 8/30/24, 6:54 AM "Doug W." &lt;<a href="mailto:webservant@aascv.org" onclick="parent.window.phx.iac.notify('mail_compose', {'to':['webservant@aascv.org']}); return false;" target="_blank">webservant@aascv.org</a>&gt; wrote:</div><blockquote class="gmail_quote" style="margin: 0.8ex 0.0pt 0.0pt 0.8ex;border-left: 1.0px solid rgb(204,204,204);padding-left: 1.0ex;">

  <div>
   Cheri:
   <div>
    <br>
   </div>
   <div>
    I'm sorry to bother you, but something changed or broke in the JSON feed <a href="https://deref-mail.com/mail/client/2-54eO6NKZk/dereferrer/?redirectUrl=https%3A%2F%2Favcentraloffice.org%2Fmap%2Fjson.cfm" target="_blank">https://avcentraloffice.org/map/json.cfm</a>,
 or the 12-Step meeting plugin update changed something for non-standard
 feeds. The SCV feed is fine. The 12-step plugin is used on the district
 WordPress site, which uses this JSON file and rejects it. The attached 
email shows that all AV meetings are out of sync, and the sync on the <a href="https://deref-mail.com/mail/client/eVgMexZ4KBs/dereferrer/?redirectUrl=http%3A%2F%2Farea93district7.org" target="_blank">area93district7.org</a> site goes through the motions but does not make the update, and it has been out of sync for the last couple of days. 
   </div>
   <div>
    <br>
   </div>
   <div>
    It does not affect the meetings in the Meeting Guide App (I checked it this morning).
   </div>
   <div>
    <br>
   </div>
   <div>
    D
   </div>
   <div>
    <br>
    <div class="gmail_quote">
     <div class="gmail_attr">
      ---------- Forwarded message ---------
      <br>
      From: <strong class="gmail_sendername">WordPress</strong> <span>&lt;<a href="mailto:wordpress@area93district7.org" onclick="parent.window.phx.iac.notify('mail_compose', {'to':['wordpress@area93district7.org']}); return false;" target="_blank">wordpress@area93district7.org</a>&gt;</span>
      <br>
      Date: Fri, Aug 30, 2024 at 1:07 AM
      <br>
      Subject: [Area 93, District 7] Data Source Changes Detected: District 7, AV
      <br>
      To: &lt;<a href="mailto:webservant@aascv.org" onclick="parent.window.phx.iac.notify('mail_compose', {'to':['webservant@aascv.org']}); return false;" target="_blank">webservant@aascv.org</a>&gt;
      <br>
     </div>
     <br>
     <br><u></u>
     <div style="margin: 0.0px;padding: 0.0px;background-color: rgb(238,238,238);width: 100.0%;">

Updated | Surrender | Sunday @ 11:00 | 5826 | No group | Fields changed: location
Updated | A.M. Attitude Modification Meeting ® | Sunday @ 06:30 | 6135 | No group | Fields changed: location
Updated | A.M. Attitude Modification Meeting ® | Monday @ 06:30 | 6136 | No group | Fields changed: location
Updated | A.M. Attitude Modification Meeting ® | Tuesday @ 06:30 | 6137 | No group | Fields changed: location
Updated | A.M. Attitude Modification Meeting ® | Wednesday @ 06:30 | 6138 | No group | Fields changed: location
Updated | A.M. Attitude Modification Meeting ® | Thursday @ 06:30 | 6139 | No group | Fields changed: location
Updated | A.M. Attitude Modification Meeting ® | Friday @ 06:30 | 6140 | No group | Fields changed: location
Updated | A.M. Attitude Modification Meeting ® | Saturday @ 06:30 | 6141 | No group | Fields changed: location
Updated | Mid-Day Modification Meeting ® | Sunday @ 12:00 | 6142 | No group | Fields changed: location
Updated | Mid-Day Modification Meeting ® | Monday @ 12:00 | 6143 | No group | Fields changed: location
Updated | Mid-Day Modification Meeting ® | Tuesday @ 12:00 | 6144 | No group | Fields changed: location
Updated | Mid-Day Modification Meeting ® | Wednesday @ 12:00 | 6145 | No group | Fields changed: location
Updated | Mid-Day Modification Meeting ® | Thursday @ 12:00 | 6146 | No group | Fields changed: location
Updated | Mid-Day Modification Meeting ® | Friday @ 12:00 | 6147 | No group | Fields changed: location
Updated | Mid-Day Modification Meeting ® | Saturday @ 12:00 | 6148 | No group | Fields changed: location
Updated | A Gathering of Women | Sunday @ 16:00 | 6149 | No group | Fields changed: location
Updated | Open Door ® | Thursday @ 09:00 | 5922 | No group | Fields changed: location
Updated | Open Door ® | Thursday @ 12:00 | 5923 | No group | Fields changed: location
Updated | Open Door ® | Thursday @ 20:00 | 5926 | No group | Fields changed: location
Updated | Open Door ® | Friday @ 09:00 | 5946 | No group | Fields changed: location
Updated | Open Door ® | Friday @ 12:00 | 5947 | No group | Fields changed: location
Updated | Open Door ® | Friday @ 18:00 | 5948 | No group | Fields changed: location
Updated | Open Door ® | Friday @ 20:00 | 5949 | No group | Fields changed: location
Updated | Open Door ® | Saturday @ 09:00 | 5966 | No group | Fields changed: location
Updated | Open Door ® | Saturday @ 12:00 | 5967 | No group | Fields changed: location
Updated | Open Door ® | Saturday @ 18:00 | 5968 | No group | Fields changed: location
Updated | Open Door ® | Saturday @ 20:00 | 5969 | No group | Fields changed: location
Updated | Open Door ® | Thursday @ 18:00 | 6168 | No group | Fields changed: location
Updated | Open Door ® | Monday @ 09:00 | 5865 | No group | Fields changed: location
Updated | Open Door ® | Monday @ 12:00 | 5866 | No group | Fields changed: location
Updated | Open Door ® | Monday @ 18:00 | 5867 | No group | Fields changed: location
Updated | Open Door ® | Monday @ 20:00 | 5868 | No group | Fields changed: location
Updated | Open Door ® | Tuesday @ 09:00 | 5883 | No group | Fields changed: location
Updated | Open Door ® | Tuesday @ 12:00 | 5884 | No group | Fields changed: location
Updated | Open Door ® | Tuesday @ 18:00 | 5885 | No group | Fields changed: location
Updated | Open Door ® | Wednesday @ 09:00 | 5905 | No group | Fields changed: location
Updated | Open Door ® | Wednesday @ 18:00 | 5906 | No group | Fields changed: location
Updated | Open Door ® | Wednesday @ 20:00 | 5907 | No group | Fields changed: location
Updated | Open Door ® | Wednesday @ 12:00 | 5983 | No group | Fields changed: location
Updated | Open Door ® | Thursday @ 16:00 | 6254 | No group | Fields changed: location
Updated | Open Door ® | Sunday @ 18:00 | 6255 | No group | Fields changed: location
Updated | Open Door ® | Sunday @ 09:00 | 6274 | No group | Fields changed: location
Updated | Open Door ® | Sunday @ 11:00 | 6275 | No group | Fields changed: location
Updated | Open Door ® | Sunday @ 20:00 | 6276 | No group | Fields changed: location
Updated | Open Door ® | Saturday @ 22:00 | 6277 | No group | Fields changed: location
Updated | Open Door ® | Tuesday @ 20:00 | 6076 | No group | Fields changed: location
Updated | Open Door ® | Friday @ 22:00 | 6078 | No group | Fields changed: location
Updated | Open Door ® | Saturday @ 15:30 | 6282 | No group | Fields changed: location
Updated | Open Participation | Tuesday @ 17:30 | 6162 | No group | Fields changed: location
Updated | Lake Hughes Men's Stag ® | Wednesday @ 19:00 | 5904 | No group | Fields changed: location
Updated | Steps in Depth | Wednesday @ 19:00 | 6164 | No group | Fields changed: location
Updated | AV Hospital @ the Alano Club | Thursday @ 19:00 | 6212 | No group | Fields changed: location
Updated | Saturday Morning Men's Stag ® | Saturday @ 09:00 | 6225 | No group | Fields changed: location
Updated | AV Young People's Meeting | Friday @ 18:00 | 6249 | No group | Fields changed: location
Updated | Saturday Speaker Meeting | Saturday @ 19:30 | 6252 | No group | Fields changed: location
Updated | Serenity Vibes | Saturday @ 07:00 | 6259 | No group | Fields changed: location
Updated | Serenity Vibes | Saturday @ 08:15 | 6260 | No group | Fields changed: location
Updated | Wed Big Book Study ® | Wednesday @ 18:00 | 6261 | No group | Fields changed: location
Updated | Little Red Book ® | Tuesday @ 19:30 | 6262 | No group | Fields changed: location

</blockquote></div><br clear="all"><div><br></div><span class="gmail_signature_prefix">-- </span><br><div class="gmail_signature"><div>Dougie W.</div></div>
</body>
</html>Take your time since all meetings are present. Since the JSON is not in the Code for Recovery format, I am not confident they can help. I did check the JSON, and I agree it is in a valid JSON format. 

d

On Fri, Aug 30, 2024 at 6:03 AM <[cherijones@cyberservices.com](mailto:cherijones@cyberservices.com)> wrote:

    The json feed seems to be ok - I am in Montana where they only pretend to have Internet so limited as to what I can do - do you know how to post this on GitHub - you can go to code [recovery.org](https://deref-mail.com/mail/client/K95h8QXX7HY/dereferrer/?redirectUrl=http%3A%2F%2Frecovery.org) for the link

    --
    Sent with [mail.com](https://deref-mail.com/mail/client/Wx6gCtsMPRc/dereferrer/?redirectUrl=http%3A%2F%2Fmail.com) Mail app from my phone
    On 8/30/24, 6:54 AM "Doug W." <[webservant@aascv.org](mailto:webservant@aascv.org)> wrote:

        Cheri:

        I'm sorry to bother you, but something changed or broke in the JSON feed [https://avcentraloffice.org/map/json.cfm](https://deref-mail.com/mail/client/2-54eO6NKZk/dereferrer/?redirectUrl=https%3A%2F%2Favcentraloffice.org%2Fmap%2Fjson.cfm), or the 12-Step meeting plugin update changed something for non-standard feeds. The SCV feed is fine. The 12-step plugin is used on the district WordPress site, which uses this JSON file and rejects it. The attached email shows that all AV meetings are out of sync, and the sync on the [area93district7.org](https://deref-mail.com/mail/client/eVgMexZ4KBs/dereferrer/?redirectUrl=http%3A%2F%2Farea93district7.org) site goes through the motions but does not make the update, and it has been out of sync for the last couple of days. 

        It does not affect the meetings in the Meeting Guide App (I checked it this morning).

        D

        ---------- Forwarded message ---------
        From: WordPress <[wordpress@area93district7.org](mailto:wordpress@area93district7.org)>
        Date: Fri, Aug 30, 2024 at 1:07 AM
        Subject: [Area 93, District 7] Data Source Changes Detected: District 7, AV
        To: <[webservant@aascv.org](mailto:webservant@aascv.org)>

        Data Source Changes Detected: District 7, AV
        Data Source changes were detected during a scheduled sychronization check with this feed: [https://avcentraloffice.org/map/json.cfm](https://deref-mail.com/mail/client/2-54eO6NKZk/dereferrer/?redirectUrl=https%3A%2F%2Favcentraloffice.org%2Fmap%2Fjson.cfm). Your website meeting list details based on the District 7, AV feed are no longer in sync.

        Please sign-in to your website and refresh the District 7, AV Data Source feed found on the Meetings Import & Export page.

        data_source_name: District 7, AV
        parent_region:
        database record count: 207
        data source feed count: 207
        Last Refresh: *Thursday August 29, 2024 05:17 am

### Relevant log output/errors

```shell
json path from the central office - http://avcentraloffice.org/map/json.cfm

nothing has changed on avcentraloffice.org

not sure if the problem is with TSML or CF - it has been working - this was set up last April
```

**Comment by joshreisner** [comment]
hey @dougie-wougie are you having trouble importing the feed from antelope valley? what happens on your end? it seems to import to my test site ok.

<img width="1083" alt="Screenshot 2024-09-16 at 12 16 25 PM" src="https://github.com/user-attachments/assets/05d574dc-2806-4b9d-aa4e-3585d80d5cd7">

**Reply by joshreisner** [reply]
hi @dougie-wougie just wanted to be sure you saw the post ☝️ - what behavior or message were you seeing when trying to import the feed?

it worked ok for me on monday - i see that right now https://avcentraloffice.org is not responding, so in wordpress i'm seeing the error `Fatal error: Maximum execution time of 30 seconds exceeded in /Users/jreisner/Herd/wordpress/wp-includes/Requests/src/Transport/Curl.php on line 205` cc @CheriJones

**Reply by dougie-wougie** [reply]
Thanks, Josh. 

As I have been told, avcentraloffice.org is working; that site does not use WordPress; it is built on ColdFusion. He is the feed I am trying to use. https://avcentraloffice.org/map/json.cfm

The issue is at area93district7.org. I am using the feeds (https://area93district7.org/wp-admin/edit.php?post_type=tsml_meeting&page=import), and it reports a moving error every day when I sync them in a nightly email. 

The district has two central offices, but the district site lists them on WordPress only for area93district7.org. aascv.org is the other, and there are no issues with that one.

An error looks like this for all 270 meetings:

Data Source Changes Detected
Please sign in to your website and refresh the District 7, AV feed on the Import & Export page.

Surrender
Sunday @ 11:00 am	Changed: Location

Local Speaker
Sunday @ 8:00 pm	Changed: Location

A.M. Attitude Modification Meeting ®
Sunday @ 6:30 am	Changed: Location


D.

**Reply by joshreisner** [reply]
oh ok thanks @dougie-wougie - i don't think this is an error, it's just reporting that it sees a change in the Location field for these meetings - the issue is something that's always existed, but TSML may be reporting it differently because of some recent changes

what's going on is TSML groups locations by address, and only saves a single `Location` and `Location Notes` for every geocoded address it sees. The AV Central Office site does not follow those rules however, so there is a discrepancy

for example, the "Surrender" meeting is at `509 West Avenue H 8` in Lancaster. mostly the location key for this address is empty, but it also sees `Near Sierra Hwy & H 8` and `Open Door`

Meeting Guide also works like TSML does, so the best solution would be to update the AV Central Office website to group meetings the same way, but that would be a heavy engineering lift

the best / quickest solution would be simply to disable change detection for AV Central Office so you don't get an email about it every night

**Reply by dougie-wougie** [reply]
Thank you, Josh:

I wish the AV central office would use WordPress and the plugin, but they do not. I offered to rebuild the site for them. 

I can only offer suggestions. I will ignore the errors and hope one day, they will upgrade. The manager likes the prebuilt "panels" for quick changes, but it limits the site in function and design.

d
